### PR TITLE
V3 Benchmarks

### PR DESCRIPTION
**Pull request by WrathfulSpatula**
This branch includes v3 release candidate benchmarks, for analysis and discussion.

### CODE REVIEW

File: docs/performance.rst [hunk 1]
```diff
@@ -47,8 +47,11 @@ engine type, (per [Pednault2017]_,) and "QFusion" provides a "gate fusion"
 layer. A "QEngine" type is always the base layer, and QUnit and QFusion types 
 may be layered over these, and over each other.
 
-Future publications will compare the performance of Qrack against other
-publicly available simulators, as rigorous implementations can be implemented.
+This version of the Qrack benchmarks contains comparisons against other
+publicly available simulators, specifically QCGPU and ProjectQ (with its
+default simulator). Note that Qrack has been incorporated into an optional
```
Comment (bennbollay) [inline docs/performance.rst]: ... incorporates as an optional ...

File: docs/performance.rst [hunk 2]
```diff
@@ -72,7 +75,11 @@ Disclaimers
...
+100 timed trials of single and parallel gates were run for each qubit count between 3 and 28 qubits. CPU and GPU benchmarks were run on two respective systems that could represent realistic use cases for each engine type. An AWS p3.2xlarge running Ubuntu Server 18.04LTS was used for GPU benchmarks. An Alienware 17 R5 with an Intel(R) Core(TM) i7-8750H running Ubuntu 18.04LTS was used for Qrack CPU benchmarks. The average and quartile boundary values of each set of 100 were recorded and graphed. Grover's search to invert a black box subroutine, or "oracle," was similarly implemented for trials between 3 and 18 qubits. Grover's algorithm was iterated an optimal number of times, vs. qubit count, to maximize probability on a half cycle of the algorithm's period, being :math:`floor\left[\frac{\pi}{4asin\left(1/\sqrt{2^N}\right)}\right]` iterations for :math:`N` qubits.
```
Comment (bennbollay) [inline docs/performance.rst]: Add links to the function (if possible) or the line number where these tests are implemented.

File: docs/performance.rst [hunk 3]
```diff
@@ -72,7 +75,11 @@ Disclaimers
...
+
+A quantum fourier transform was run for 100 trials between 3 and 26 qubits on the GPU engine types. ProjectQ and QCGPU were also benchmarked for this test, on an AWS p3.2xlarge running Ubuntu Server 18.04LTS, with the benchmarking code provided by QCGPU, in python. The benchmarking script from QCGPU iterates at random between the engine options. 200 total trials were run for the set of QCGPU and ProjectQ, selected by approximately equal probability random chance in each of the 200 iterations.
```
Comment (bennbollay) [inline docs/performance.rst]: python should be capitalized, and the version of QCGPU and Python specified.

File: docs/performance.rst (hunk 2 above)
Comment (bennbollay) [reply docs/performance.rst]: Also, be sure to specify the date that these results were generated somewhere.

File: docs/performance.rst (hunk 1 above)
Comment (WrathfulSpatula) [reply docs/performance.rst]: Done.

File: docs/performance.rst (hunk 2 above)
Comment (WrathfulSpatula) [reply docs/performance.rst]: Both done.

File: docs/performance.rst (hunk 3 above)
Comment (WrathfulSpatula) [reply docs/performance.rst]: Done. The date that QCGPU was accessed is now given.

**Review by bennbollay — APPROVED** [review]

**Pull request merged**